### PR TITLE
VIR-23: Position modify view

### PR DESCRIPTION
**Pull request by Peskarhu**
* User can now modify the position data and submit. Once submitted, the
   value is loaded onto the screen
* Uses lesser role due to testing purposes. NEED TO CHANGE LATER!
* Add updateposition data transfer object so that only updated values
   are added to the position data
* Small improvements to the code and usability
* Feature for closing details view
* Already clicked details view/position can be closed if open
* Small issue happens when the position is modified and submitted as the
   view closes
* UI design fixes and improvements
* Add all of the required fields with correct settings
* Made small adjustments to the csv modal and create modal to unify each
  modal
* Add translations

### CODE REVIEW

**Comment by Peskarhu** [comment]
Näyttökuva 2024-11-29

**Comment by Peskarhu** [comment]
kuva

**Comment by Peskarhu** [comment]
kuva

**Comment by Peskarhu** [comment]
Some small usability improvements. The user can see the old value as a reminder/placeholder if the field is empty: 
kuva

File: backend/Virkarekisteri/src/Functions/Positions/UpdatePosition.cs [hunk 1]
```diff
@@ -24,68 +24,56 @@ IPositionNameRepository positionNameRepository
     /// No content on success or an error
     /// </returns>
     [Function("UpdatePosition")]
-    [RequiresEditRole]
+    [RequiresReadRole] // <--- this should be admin or edit!! For testing purposes, I use lesser role
```
Comment (joonasyr) [inline backend/Virkarekisteri/src/Functions/Positions/UpdatePosition.cs]: This should be changed back to edit or admin

**Review by joonasyr — CHANGES_REQUESTED** [review]
Seems okay besides this one change

File: backend/Virkarekisteri/src/Functions/Positions/UpdatePosition.cs (hunk 1 above)
Comment (Peskarhu) [reply backend/Virkarekisteri/src/Functions/Positions/UpdatePosition.cs]: Yep, forgot that

**Comment by Peskarhu** [comment]
Note: The Päätös is currently not added to the database. This was intentional, as I wanted to avoid potential conflicts if Joona has already implemented a solution for Päätös for the history/changeLog.

**Review by joonasyr — APPROVED** [review]

**Pull request merged**